src/UserAction/StepExecutorTrait.php

```diff
@@ -72,6 +72,9 @@ trait StepExecutorTrait
     /** @var string */
     public $finalMsg = 'Complete!';
 
+    /** @var array An extended copy of UserAction arguments. It contains original action arguments and arguments set by '__atk_model'. */
+    private $cloneArgs;
```

Comment:
I think we should add native Model args support to atk4/data model. It is also not completely clear to me, why there are args & fields. In atk4/data, the UserAction::$fields property is described to be used for dirty detection only.